Don't skip building a chroot for non-strict non-package cooks. (RMK-931)

diff -r 0403d9beba9c NEWS
--- a/NEWS	Thu Nov 13 17:56:22 2008 -0500
+++ b/NEWS	Mon Nov 17 19:16:56 2008 +0000
@@ -1,3 +1,11 @@
+Changes in @NEW@:
+ o Bug Fixes:
+     * A chroot is now created for all builds, even with strictMode set
+       to False. Previously, the fake chroot used in this scenario
+       caused group cooks to fail with Conary 2.0.29 and later.
+       (RMK-931)
+
+
 Changes in 1.0.24:
   o Build Changes:
      * Recipes are now loaded in the worker as a command instead of
diff -r 0403d9beba9c rmake/worker/chroot/rootfactory.py
--- a/rmake/worker/chroot/rootfactory.py	Thu Nov 13 17:56:22 2008 -0500
+++ b/rmake/worker/chroot/rootfactory.py	Mon Nov 17 19:16:56 2008 +0000
@@ -347,18 +347,6 @@
                 ' please remove the old root by hand.' % root)
         return not removeFailed
 
-class FakeRmakeRoot(rMakeChroot):
-    def canChroot(self):
-        return False
-
-    def checkSanity(self):
-        pass
-
-    def install(self):
-        pass
-
-    def useStandardRoot(self):
-        return False
 
 class ExistingChroot(rMakeChroot):
     def __init__(self, rootPath, logger, chrootHelperPath):
diff -r 0403d9beba9c rmake/worker/chroot/rootmanager.py
--- a/rmake/worker/chroot/rootmanager.py	Thu Nov 13 17:56:22 2008 -0500
+++ b/rmake/worker/chroot/rootmanager.py	Mon Nov 17 19:16:56 2008 +0000
@@ -195,19 +195,8 @@
         if not setArch:
             targetArch = None
 
-        if (not cfg.strictMode and not buildTrove.isPrepOnly() and
-            (buildTrove.isRedirectRecipe() or buildTrove.isGroupRecipe()
-               or buildTrove.isFilesetRecipe())):
-            # we don't need to actually instantiate a root to cook
-            # these packages - if we're not worried about using the 
-            # "correct" conary, conary-policy, etc.
-            jobList = []
-            util.mkdirChain(self.baseDir)
-            os.chdir(self.baseDir)
-            chrootClass = rootfactory.FakeRmakeRoot
-        else:
-            chrootClass = rootfactory.FullRmakeChroot
-            util.mkdirChain(self.baseDir)
+        chrootClass = rootfactory.FullRmakeChroot
+        util.mkdirChain(self.baseDir)
         copyInConary = (not targetArch
                         and not cfg.strictMode
                         and cfg.copyInConary)
